gdeep/trainer/trainer.py

```diff
@@ -24,6 +34,9 @@
 from gdeep.models import ModelExtractor
 from gdeep.utility import _inner_refactor_scalars
 from gdeep.utility import DEVICE
+
+if TYPE_CHECKING:
```

Comment:
Why is this needed?

Reply:
avoid circular import

Reply:
Since `Trainer` is not needed anymore in `regularizer.py`, you don't have circular imports anymore.